coordinator/src/main/scala/filodb.coordinator/NodeCoordinatorActor.scala

```diff
@@ -124,16 +132,20 @@ class NodeCoordinatorActor(metaStore: MetaStore,
   private def setupIngestion(originator: ActorRef,
                              dataset: DatasetRef,
                              columns: Seq[String],
-                             version: Int): Unit = {
+                             version: Int,
+                             reloadFlag: Boolean = false): Unit = {
     def notify(msg: Any): Unit = { self ! DatasetCreateNotify(dataset, version, msg) }
-
     def createDatasetCoordActor(datasetObj: Dataset, richProj: RichProjection): Unit = {
-      val props = DatasetCoordinatorActor.props(richProj, version, columnStore, reprojector, config)
+      val props = DatasetCoordinatorActor.props(richProj, version, columnStore,
+                                                reprojector, actorPath, config, reloadFlag)
       val ref = context.actorOf(props, s"ds-coord-${datasetObj.name}-$version")
-      self ! AddDatasetCoord(dataset, version, ref)
-      notify(IngestionReady)
+      self ! AddDatasetCoord(originator,dataset, version, ref, reloadFlag)
+      if (!reloadFlag) {
+        val colDefinitions =richProj.columns.foldLeft("")(_  + _.toString + ";").dropRight(1)
```

Comment:
You should reuse the code in DatasetTable.scala - see stringsToStr - using \x01 as a split char makes it more visible in cqlsh.   Basically `richProj.columns.map(_.toString).mkString('\x01')`.  Also be ware that RichProjection.columns includes computed columns, which you probably want to leave out.